src/View/XmlView.php

```diff
@@ -133,10 +133,10 @@ protected function _serialize(array|string $serialize): string
                 }
             }
         } else {
-            /** @var array<mixed>|string $data */
+            /** @var array<mixed>|string|int|bool|null $data */
             $data = $this->viewVars[$serialize] ?? [];
             if (
-                $data &&
+                $data !== null &&
```

Comment:
Do we have a test where `$data` ends up being `[]`, as the semantics have changed here a bit?

Reply:
We don't, I can add a scenario for that.

Reply:
Added a test for `[]` as well. Prior to this change, `[]` would also error.